core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -606,13 +616,15 @@ void maybeProcessFetchQueue() {
                 });
             });
 
-            Set<DelayedShareFetchKey> delayedShareFetchWatchKeys = new HashSet<>();
+            Set<Object> delayedShareFetchWatchKeys = new HashSet<>();
             shareFetchData.partitionMaxBytes().keySet().forEach(
-                topicIdPartition -> delayedShareFetchWatchKeys.add(
-                    new DelayedShareFetchKey(shareFetchData.groupId(), topicIdPartition)));
+                topicIdPartition -> {
+                    delayedShareFetchWatchKeys.add(new DelayedShareFetchKey(shareFetchData.groupId(), topicIdPartition));
+                    delayedShareFetchWatchKeys.add(topicIdPartition);
+                });
 
             // Add the share fetch to the delayed share fetch purgatory to process the fetch request.
-            addDelayedShareFetch(new DelayedShareFetch(shareFetchData, replicaManager, partitionCacheMap, delayedActionsQueue, delayedShareFetchPurgatory),
+            addDelayedShareFetch(new DelayedShareFetch(shareFetchData, replicaManager, partitionCacheMap, this),
```

Comment:
Will it be wise to send a function which can be executed instead of reference of `share parition manager`?

Reply:
I think its a better option to send a reference of `SharePartitionManager` because we need it for 2 purposes right now - 
1. call `addPurgatoryCheckAndCompleteDelayedActionToActionQueue`
2. perform a get on `partitionCacheMap` (changes done in my latest commit)

Plus, in the future, when we move the code for share partition initialization to `DelayedShareFetch` from SPM, we will need to update the `partitionCacheMap` as well, so we'll be needing a function to do that from `DelayedShareFetch`. 
Considering all these utilities and future integrations, I feel we can let it remain like this and change it in the future, if we feel otherwise.